File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/StatisticConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiStatistic;
import com.platdmit.simplecloudmanager.data.database.entity.DbStatistic;
import com.platdmit.simplecloudmanager.domain.models.Statistic;

import java.util.List;

public interface StatisticConverter {
    DbStatistic fromApiToDb(ApiStatistic apiStatistic, long serverId);
    Statistic fromDbToDomain(DbStatistic dbStatistic);
    List<DbStatistic> fromApiToDbList(List<ApiStatistic> apiStatistics, long serverId);
    List<Statistic> fromDbToDomainList(List<DbStatistic> dbStatistics);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/StatisticConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.simplecloudmanager.data.api.models.ApiStatistic
import com.platdmit.simplecloudmanager.data.database.entity.DbStatistic
import com.platdmit.simplecloudmanager.domain.models.Statistic

interface StatisticConverter {
    fun fromApiToDb(apiStatistic: ApiStatistic, serverId: Long): DbStatistic
    fun fromDbToDomain(dbStatistic: DbStatistic): Statistic
    fun fromApiToDbList(apiStatistics: List<ApiStatistic>, serverId: Long): List<DbStatistic>
    fun fromDbToDomainList(dbStatistics: List<DbStatistic>): List<Statistic>
}